parser.js

```diff
@@ -0,0 +1,84 @@
+const getMarkdown = (str, re, tag) => {
+	const result = str.match(re);
+
+	if (result && result.length) {
+		return `<${tag}>${result[0]}</${tag}>`;
+	}
+
+	return str;
+};
+
+const rules = [
+	{
+		name: 'Heading level 1',
+		tag: 'h1',
+		regex: /(?<=^# )(.*)/,
+		parse: getMarkdown,
+	},
+	{
```

Comment:
I would suggest that you try to encapsulate what each element does into a separate module, this will help you making the code easier to extend.

An element could be defined with a class, but this could be also a module or whatever (the important part is that all related functionality is close together):

```
class H1 {
  constructor(str) {
    this.rawText = str;
  }

  static matchesElement(str) {
    return str.matches(/(?<=^- )(.*)/);
  }

  get text() {
    // gets the raw text without any markdown
  }

  renderHTML() {
     // uses this.text to render HTML
  }
}
```

The advantage of doing something like this is that extending the system with new elements is much easier, you just create a new class that complies with the same interface as other elements, and it will just work.